api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
@ironcladlou I think the separation makes sense but I wonder how @derekparker feels, since he may not be familiar w/ that pattern.